### PR TITLE
Update .gitignore

### PR DESCRIPTION
**Pull request by trepidity**
Added several common ignore statements. They may not all be needed, but they will not hurt anything.

Would anyone care to comment or approve this PR for me? Thanks.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -24,3 +24,34 @@ DerivedData
...
+
+# Watchman
+.watchmanconfig
+
+# Android
+.idea
+.gradle
+local.properties
+*.iml
+/android/
```
Comment (talha-ahmed) [inline .gitignore]: I am wondering if this has broken android? I don't get android folder in my node_modules/react-native-sound after `npm install react-native-sound --save` hence the `react-native link react-native-sound` fails to link android and only links iOS and consequently getting RNSound.IsAndroid undefined.

Comment (trepidity) [reply .gitignore]: YIKES!!
Not sure how that one android folder made it in. I'm going to revert the change.

**Comment by benevbright** [comment]
Is this published?
Thanks.

**Comment by trepidity** [comment]
I reverted it. We need to make a new PR with suggested values.

**Comment by trepidity** [comment]
@benevbright checkout #400

**Comment by benevbright** [comment]
Thanks!

**Pull request merged**